### PR TITLE
CORE-8184: Remove hardcoded HTTP RPC password from CI's Helm chart

### PR DESCRIPTION
**Pull request by vkolomeyko**
Also changes are made to multi-clustered E2E tests to provision for distinct HTTP RPC passwords in different clusters.

### CODE REVIEW

File: .ci/e2eTests/corda.yaml [hunk 1]
```diff
@@ -2,9 +2,6 @@ imagePullSecrets:
   - "docker-registry-cred"
 
 bootstrap:
-  initialAdminUser:
-    password:
-      value: "admin"
```
Comment (vkolomeyko) [inline .ci/e2eTests/corda.yaml:7]: The "meat" of the change whereby by removing hardcoded value we will rely on random generation of the password and storing it as a secret.
**Note:** Different clusters will have different values generated and therefore different passwords.

File: applications/workers/release/rpc-worker/src/e2eTest/kotlin/net/corda/applications/workers/rpc/utils/E2eCluster.kt [hunk 2]
```diff
@@ -1,21 +1,30 @@
...
 import net.corda.applications.workers.rpc.kafka.KafkaTestToolKit
+import net.corda.httprpc.RpcOps
+import net.corda.httprpc.client.HttpRpcClient
 
 interface E2eCluster {
     val members: List<E2eClusterMember>
 
     val clusterConfig: E2eClusterConfig
 
-    val testToolkit: TestToolkit
```
Comment (vkolomeyko) [inline applications/workers/release/rpc-worker/src/e2eTest/kotlin/net/corda/applications/workers/rpc/utils/E2eCluster.kt:12]: Toolkit is no longer accessible directly from `E2eCluster`, however all the required methods/properties been introduced to the `interface E2eCluster`.

File: .ci/e2eTests/Jenkinsfile [hunk 3]
```diff
@@ -165,6 +165,15 @@ pipeline {
             }
         }
         stage('Forward port and run tests') {
+            environment {
+                E2E_CLUSTER_A_RPC_PASSWORD = "${getInitialAdminPassword('alice')}"
+                E2E_CLUSTER_B_RPC_PASSWORD = "${getInitialAdminPassword('bob')}"
+                E2E_CLUSTER_C_RPC_PASSWORD = "${getInitialAdminPassword('caroline')}"
```
Comment (davidcurrie) [inline .ci/e2eTests/Jenkinsfile:171]: nit: is the `"${get...}"` redundant and you could just have `get...`?

**Review by davidcurrie — APPROVED** [review]
LGTM - $1M question is then whether we also remove the hard-coded `admin` password from the developer `values.yaml` at the root of the repo and update the instruction for how to set the environment variable when running the E2E tests?

**Review by charlieR3 — APPROVED** [review]

**Review by yift-r3 — APPROVED** [review]

File: .ci/e2eTests/Jenkinsfile (hunk 3 above)
Comment (vkolomeyko) [reply .ci/e2eTests/Jenkinsfile:171]: I was just following an existing pattern in this file where people call method in Groovy like: `"${methodCall()}"`.
I do not mind to make a change, but just do not want to trigger another round of CI/approvals for this minor change.

**Comment by vkolomeyko** [comment]
> LGTM - $1M question is then whether we also remove the hard-coded `admin` password from the developer `values.yaml` at the root of the repo and update the instruction for how to set the environment variable when running the E2E tests?

I would leave `admin` password in `values.yaml` as-is TBH.
If a crafty cluster deployer wants to remove this default value - they will get a cluster with random password. But since they are crafty/knowlegable, they will know the way how to obtain the password from K8s secrets.

For simpler people, I would rather not change the way they currently work with local K8s deployment and leave `admin/admin` logon working as before. Especially given that combined worker is also using the same credentials and there is no way to use K8s secrets in this case.

**Review by benmcmahon100 — APPROVED** [review]
LGTM

**Comment by davidcurrie** [comment]
> For simpler people, I would rather not change the way they currently work with local K8s deployment and leave admin/admin logon working as before.

I think that is not unreasonable. My concern is that people take this `values.yaml` as the starting point for other deployments. I'll give it some thought but I think it may be worth at least strengthening the statement at the top of that file to reinforce that these values are intended for development usage.

**Pull request merged**